chart/templates/deployment.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: "{{ .Release.Name }}"
+  name: "{{ .Release.Name }}"
+  namespace: "{{ .Release.Namespace }}"
+  annotations:
+    {{- if .Values.iam_role }}
+    iam.amazonaws.com/role: {{ .Values.iam_role }}
+    {{- end }}
```

Comment:
IAM role should be specified in the service account, as shown here https://docs.aws.amazon.com/eks/latest/userguide/specify-service-account-role.html